### PR TITLE
fix(quota): return unlimited free space for type changes.

### PR DESCRIPTION
**Pull request by camilasan**
Close https://github.com/nextcloud/desktop/issues/8555

### CODE REVIEW

File: src/common/syncjournaldb.cpp [hunk 1]
```diff
@@ -851,8 +855,8 @@ bool SyncJournalDb::updateMetadataTableStructure()
 
     addColumn(QStringLiteral("isLivePhoto"), QStringLiteral("INTEGER"));
     addColumn(QStringLiteral("livePhotoFile"), QStringLiteral("TEXT"));
-    addColumn(QStringLiteral("quotaBytesUsed"), QStringLiteral("BIGINT"));
-    addColumn(QStringLiteral("quotaBytesAvailable"), QStringLiteral("BIGINT"));
+    addColumn(QStringLiteral("quotaBytesUsed"), QStringLiteral("BIGINT"), false, QStringLiteral("DEFAULT -1 NOT NULL"));
+    addColumn(QStringLiteral("quotaBytesAvailable"), QStringLiteral("BIGINT"), false, QStringLiteral("DEFAULT -1 NOT NULL"));
```
Comment (mgallien) [inline src/common/syncjournaldb.cpp]: this contradicts with the lines 75-76 which do test for `NULL` values
I would advice to have one way or the other to decrease complexity
so either you add `NOT NULL` for the column or you test for `NULL` values

**Review by mgallien — CHANGES_REQUESTED** [review]
I realize that our testing coverage is probably not good enough because you did not modified any test
work to be done later I guess

File: src/common/syncjournaldb.cpp [hunk 2]
```diff
@@ -742,6 +747,58 @@ bool SyncJournalDb::updateMetadataTableStructure()
...
+
+        // the column already exists, check if default has been set
+        if (setDefault) {
+            SqlQuery query(_db);
+            const auto selectDefault = QStringLiteral("SELECT dflt_value FROM pragma_table_info('metadata') WHERE name = '%1';").arg(columnName);
+            query.prepare(selectDefault.toLatin1());
+            if (!query.exec()) {
+                sqlFail(QStringLiteral("check default value for: %1").arg(columnName), query);
+                re = false;
+            }
```
Comment (nilsding) [inline src/common/syncjournaldb.cpp]: shouldn't there be a `return` here as well?
```suggestion
            if (!query.exec()) {
                sqlFail(QStringLiteral("check default value for: %1").arg(columnName), query);
                re = false;
                return;
            }
```

File: src/common/syncjournaldb.cpp [hunk 3]
```diff
@@ -742,6 +747,58 @@ bool SyncJournalDb::updateMetadataTableStructure()
...
+
+            if (const auto result = query.next();result.ok && result.hasData && !query.nullValue(0)) {
+                qCWarning(lcDb) << "column" << columnName << "has default value of" << query.intValue(0);
```
Comment (nilsding) [inline src/common/syncjournaldb.cpp]: since this should be the more common case, change the info level to `Debug`
 
```suggestion
                qCDebug(lcDb) << "column" << columnName << "has default value of" << query.intValue(0);
```

**Review by nilsding — CHANGES_REQUESTED** [review]

File: src/common/syncjournaldb-copy.cpp
Comment (mgallien) [inline src/common/syncjournaldb-copy.cpp:1]: I am not sure why you are adding new files

File: src/common/syncjournaldb-copy.h
Comment (mgallien) [inline src/common/syncjournaldb-copy.h:1]: I am not sure why you are adding new files

File: test/testsyncjournaldb.cpp [hunk 4]
```diff
@@ -98,6 +98,52 @@ private slots:
...
+        QCOMPARE(storedRecord._folderQuota.bytesUsed, 100);
+    }
+
+    void testFolderMigration() {
+        const auto quotaBytesUsed =  QStringLiteral("quotaBytesUsed");
+        const auto quotaBytesAvailable =  QStringLiteral("quotaBytesAvailable");
+        const auto metadata = QStringLiteral("metadata");
+        const auto columnsBeforeRemoval = _db.tableColumns(metadata.toLatin1());
+        QVERIFY(columnsBeforeRemoval.indexOf(quotaBytesUsed.toLatin1()) > -1);
+        QVERIFY(columnsBeforeRemoval.indexOf(quotaBytesAvailable.toLatin1()) > -1);
```
Comment (nilsding) [inline test/testsyncjournaldb.cpp]: nit: for comparisons like these I'd prefer the usage of [`QCOMPARE_GT(computed, baseline)`](https://doc.qt.io/qt-6/qtest.html#QCOMPARE_GT) as it'll provide a better error message in case of a failing check
 
```suggestion
        QCOMPARE_GT(columnsBeforeRemoval.indexOf(quotaBytesUsed.toLatin1()), -1);
        QCOMPARE_GT(columnsBeforeRemoval.indexOf(quotaBytesAvailable.toLatin1()), -1);
```

**Review by nilsding — APPROVED** [review]

**Review by mgallien — APPROVED** [review]
